pkg/cache/controller.go

```diff
@@ -37,8 +36,12 @@ func ClusterIndexFunc(obj interface{}) ([]string, error) {
 	if err != nil {
 		return []string{}, fmt.Errorf("object has no meta: %v", err)
 	}
-	clusterName := logicalcluster.From(meta).String()
-	return []string{ToClusterAwareKey(clusterName, "", "")}, nil
+	return []string{ClusterIndexKey(logicalcluster.From(meta))}, nil
+}
+
+// ClusterIndexKey formats the index key for a cluster name
+func ClusterIndexKey(clusterName logicalcluster.Name) string {
```

Comment:
Exposing a clear index-formatter func allows the user to know exactly what types are required to use the index.